src/actions/sign_up_confirmations/show.cr

```diff
@@ -0,0 +1,26 @@
+class SignUpConfirmations::Show < BrowserAction
+  include Auth::AllowGuests
+
+  get "/confirm/:token" do
+    user = UserQuery.new.confirmation_token(token).first?
+    flash.keep
+
+    if user && user.confirmed?
+      flash.info = "You've already confirmed your email! Please sign in."
+      redirect to: SignIns::New
+    elsif user && !user.confirmed?
```

Comment:
This might seem small but what do you think about adding a method instead of using `!`.
Something like.
```suggestion
    elsif user && user.unconfirmed?
```

Reply:
That's a great idea, @wontruefree, will do!

I have yet to play with whatever the Crystal/Lucky equivalent of Rails concerns are, but maybe I should extract the columns and helpers into a `User::EmailConfirmable` module or something to keep the more "important" logic front and center in the model.